model/lifecycle.go

```diff
@@ -444,11 +445,19 @@ func RestartVersion(versionId string, taskIds []string, abortInProgress bool, ca
 	taskGroupsToCheck := map[taskGroupAndBuild]task.Task{}
 	tasksToRestart := finishedTasks
 	if abortInProgress {
-		tasksToRestart, err = task.Find(task.ByIds(taskIds))
+		aborted, err := task.Find(task.BySubsetAborted(taskIds))
```

Comment:
this change handles the case when some tasks might be running while the commit queue dequeues something in front. We want them to finish running then reset. Resetting them immediately strands the existing task

Reply:
I'm confused about this change. Didn't the task already get aborted? So what needs to finish running? 
> when some tasks might be running while the commit queue dequeues something in front. 
Can you illustrate this a bit more? Do you mean tasks are running for later items in the queue, but an earlier item failed or was removed? I don't understand what is stranded.

Does this change the behavior of restarting versions from the UI and aborting in progress tasks? We would want those to be aborted and restarted immediately.

Reply:
Also, these are marked as SetResetWhenFinished, aren't they still added to the restartIds at 473? And why do we need a query to get Aborted tasks, is that field not set in memory? It seems like this if could be rolled into loop at 462.

Reply:
if tasks get restarted immediately after they're aborted, they get stranded because the fields that identify the host-task pairing get cleared, and the first execution can't end because the app server rejects any requests from that host due to the task-host check failing. This is true for all tasks, but we just haven't had any workflows that needed to automatically restart an aborted task before I think. I'm querying for these specifically because I just want the ones that were aborted on line 408, but there isn't a way to run an update and have that query return you the ones it updated (just the count)

Reply:
We don't have any workflows that automatically restart an aborted task, but we have the Restart Version panel, which also restarts In Progress tasks. But yeah it doesn't seem to work right.

What about this question? All the tasks in aborted are still going to be added to restartIds. Should they not be?
> Also, these are marked as SetResetWhenFinished, aren't they still added to the restartIds at 473?

Reply:
Restarting from the UI goes through the same code path and has the same problem. I think what makes it less noticeable is that the older execution is what gets stuck, but even that gets cleaned up by the stranded task job. I just kept noticing it because I was looking for errors in staging. They specifically aren't added to restartIds because we don't want to restart them right now

Reply:
OH I didn't see that tasksToRestart were only finishedTasks.  I am now following.